Formula/solid.rb

```diff
@@ -15,11 +17,19 @@ class Solid < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "8cb9ab57cd86c4d1b5da960ea243a776e1b7a695652bf3cbced579bfcf01c312"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
   # This patch fixes a broken build on clang-600.0.56.
   # Was reported to bugs@dtecta.com (since it also applies to solid-3.5.6)
   patch :DATA
 
   def install
+    # Avoid `required file not found` errors
+    touch ["AUTHORS", "ChangeLog", "NEWS"]
```

Comment:
We probably need to touch the `info` pages too to avoid this trying to regenerate them.

Reply:
The existing `touch` call creates a few files that `configure` wants, otherwise we don't reach `make install`. The `makeinfo` failure on Linux happens during `make install`, so the issues are a bit different. Adding `"doc"`, `"doc/*"`, or both to the `touch` array doesn't resolve this Linux issue, if that's in line with what you intended.

The related error suggests this may be related to `make`:

```
WARNING: 'makeinfo' is missing on your system.
         You should only need it if you modified a '.texi' file, or
         any other file indirectly affecting the aspect of the manual.
         You might want to install the Texinfo package:
         <https://www.gnu.org/software/texinfo/>
         The spurious makeinfo call might also be the consequence of
         using a buggy 'make' (AIX, DU, IRIX), in which case you might
         want to install GNU make:
         <https://www.gnu.org/software/make/>
```

I tried this using the `make` formula on Linux (i.e., `depends_on "make" => :build` and `ENV.prepend_path "PATH", Formula["make"].opt_libexec/"gnubin"` in the `install` block) but it didn't make a difference. It builds fine using `cmake` on Linux (fails on macOS) but it doesn't produce `share/info/solid3.info` like the macOS build and existing Linux bottle. It's also worth noting that the existing `lib` file is `libsolid.so.0.0.0` but `cmake` produces `libsolid3.so.3.5.8`.

This appears to only affect Linux because the macOS build is simply picking up system `makeinfo`, so `uses_from_macos "texinfo" => :build` is one way to resolve this. This builds fine on Linux and the resulting artifacts are in line with the 3.5.6 bottle (granted `solid3.info` formatting varies due to differing `makeinfo` behavior between version 4.8 (macOS system) and brewed 6.8). [The `solid3.info` file in `solid--3.5.6.x86_64_linux.bottle.1.tar.gz` is built with `makeinfo` 4.8, which is a bit unexpected.]

Thoughts on this?

Reply:
```
You should only need it if you modified a '.texi' file, or
any other file indirectly affecting the aspect of the manual.
```
I'm thinking this means that one of the files the `touch` call creates is actually a pre-requisite of the info pages, causing them to be regenerated. We shouldn't need `texinfo` here, I think.

I suspect either `touch`ing any `*.texi` files along with any associated docs OR setting `mtime: 0` in the existing `touch` call might allow us to avoid the `texinfo` dependency.

Reply:
Hm, ok. We can depend on `texinfo` then.

Reply:
There are differences in the `solid3.info` file between macOS system `texinfo` and brewed `texinfo` (whitespace, backticks vs. single quotes, text wrapping) but we can get away with `uses_from_macos "texinfo" => :build` unless it would benefit bottling to always use brewed `texinfo`. I've updated this to use `uses_from_macos` for now.

Reply:
`uses_from_macos` is fine.